Sources/TuistGenerator/Linter/ProjectLinter.swift

```diff
@@ -60,13 +60,35 @@ class ProjectLinter: ProjectLinting {
             .reduce(into: [String: Int]()) { $0[$1] = ($0[$1] ?? 0) + 1 }
             .filter { $0.value > 1 }
             .keys
+
+        let duplicatedTargetProductNames = project.targets
+            .reduce(into: [String: Int]()) {
+                let destinations = $1.destinations.sorted(by: { a, b in
+                    a.rawValue < b.rawValue
+                }).map(\.rawValue).joined(separator: ",")
```

Comment:
I don't think we should check destinations. As per @kwridan's comment [here](https://github.com/tuist/tuist/issues/6118#issuecomment-2059658504), it's fine to have duplicate product names as long as they don't end up being linked to the same target. Sure, supporting multiple destinations was one of the use cases for this, but there might be different valid scenarios that we'd want to allow.

Ideally, we should check for all [rootTargets](https://github.com/tuist/tuist/blob/main/Sources/TuistCore/Graph/GraphTraverser.swift#L33) whether any of their [transitive dependencies](https://github.com/tuist/tuist/blob/main/Sources/TuistCore/Graph/GraphTraverser.swift#L108) has a duplicate product name.

Reply:
Thanks for the clarification! I will update and push when I have a chance.